src/test/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAOTest.java

```diff
@@ -729,6 +730,76 @@ void testDeleteStudy() throws Exception {
 
   }
 
+  @Test
+  void testExecuteUpdateDatasetWithNullName() throws Exception {
+    // This creates a study with a single dataset:
+    Study study = createStudy(List.of());
+    List<Dataset> datasets = datasetDAO.findDatasetsByIdList(study.getDatasetIds().stream().toList());
+    Dataset dataset = datasets.get(0);
+    jdbi.useHandle(handle -> serviceDAO.executeUpdateDatasetWithFiles(
+        handle,
+        dataset.getDatasetId(),
+        null,
+        study.getCreateUserId(),
+        dataset.getDacId(),
+        List.of(),
+        List.of(),
+        false)
+    );
+    Dataset updatedDataset = datasetDAO.findDatasetById(dataset.getDatasetId());
+    assertNotNull(updatedDataset);
+    assertNotNull(updatedDataset.getName());
+    List<DatasetAudit> audits = datasetDAO.findAuditsByDatasetId(dataset.getDatasetId());
+    assertFalse(audits.isEmpty());
+  }
+
+  @Test
+  void testExecuteUpdateDatasetWithEmptyName() throws Exception {
+    // This creates a study with a single dataset:
+    Study study = createStudy(List.of());
+    List<Dataset> datasets = datasetDAO.findDatasetsByIdList(study.getDatasetIds().stream().toList());
+    Dataset dataset = datasets.get(0);
+    jdbi.useHandle(handle -> serviceDAO.executeUpdateDatasetWithFiles(
+        handle,
+        dataset.getDatasetId(),
+        "",
+        study.getCreateUserId(),
+        dataset.getDacId(),
+        List.of(),
+        List.of(),
+        false)
+    );
+    Dataset updatedDataset = datasetDAO.findDatasetById(dataset.getDatasetId());
+    assertNotNull(updatedDataset);
+    assertNotNull(updatedDataset.getName());
+    List<DatasetAudit> audits = datasetDAO.findAuditsByDatasetId(dataset.getDatasetId());
+    assertFalse(audits.isEmpty());
+  }
+
+  @Test
+  void testExecuteUpdateDatasetWithNewName() throws Exception {
+    // This creates a study with a single dataset:
+    Study study = createStudy(List.of());
+    List<Dataset> datasets = datasetDAO.findDatasetsByIdList(study.getDatasetIds().stream().toList());
+    Dataset dataset = datasets.get(0);
+    String newName = RandomStringUtils.secureStrong().nextAlphabetic(dataset.getName().length() + 10);
```

Comment:
What is the purpose of using a random string here? Is it possible for this test to collide with another test?

If you do need a random string here, why use `secureStrong()` instead of `insecure()`?

If the real need is to avoid collisions, I would use a name based on a timestamp instead of a random string.

Reply:
Using random strings is a general testing pattern that we have used because we have seen flaky tests with non-random strings. With the update to org.apache.commons.lang3.RandomStringUtils, the old way we did this was deprecated, so we moved to `secureStrong` which was recommended by Elliot. I'll update this PR to make use of what we did in https://github.com/DataBiosphere/consent/pull/2429/files#diff-e141aaccd3526037b702f172e86b30f409866db7d35266121ab09ef38adf3693R182